### PR TITLE
Kotlin Migration : Create account screen

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

- Migrated create account screen (which contains personal/pro options) to Kotlin/ConstraintLayout
- Created a new build config `KOTLIN_CREATE_ACCOUNT_MIGRATION` to enable disable the feature.
- Updated styling for buttons and textviews to support different font family

https://wearezeta.atlassian.net/browse/AN-6670

#### APK
[Download build #1298](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1298/artifact/build/artifact/wire-dev-PR2624-1298.apk)
[Download build #1303](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1303/artifact/build/artifact/wire-dev-PR2624-1303.apk)
[Download build #1304](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1304/artifact/build/artifact/wire-dev-PR2624-1304.apk)
[Download build #1307](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1307/artifact/build/artifact/wire-dev-PR2624-1307.apk)
[Download build #1310](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1310/artifact/build/artifact/wire-dev-PR2624-1310.apk)
[Download build #1311](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1311/artifact/build/artifact/wire-dev-PR2624-1311.apk)
[Download build #1330](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1330/artifact/build/artifact/wire-dev-PR2624-1330.apk)

### CODE REVIEW

File: app/src/main/res-v2/layout/fragment_create_account.xml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        android:orientation="vertical"
+        android:paddingStart="@dimen/create_account_layout_container_padding_start"
+        android:paddingEnd="@dimen/create_account_layout_container_padding_end"
+        app:layout_constraintBottom_toTopOf="@+id/centerGuideline"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.9">
+
+        <com.waz.zclient.ui.text.GlyphTextView
```
Comment (mejdoo) [inline app/src/main/res-v2/layout/fragment_create_account.xml]: I will get rid of Glyphs once I receive an update from our new designer Lis. I can't download images from this screen in Zeplin right now.

File: app/src/main/res-migration/values/styles_button.xml [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
+    <style name="Button.RobotoMedium">
+        <item name="android:fontFamily">sans-serif-medium</item>
+    </style>
+
+    <style name="Button.RobotoBold">
+        <item name="android:fontFamily">sans-serif</item>
+        <item name="android:textStyle">bold</item>
+    </style>
+
+    <style name="Button.RobotoLight.Settings">
```
Comment (BradleyWire) [inline app/src/main/res-migration/values/styles_button.xml]: Can we not keep it as Button.Settings and simply inherit from the RobotoLight parent? I don't agree with the idea of the style naming having knowledge of the lower level details. From a developers perspective it should describe what the component is, not what it has.

File: app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala [hunk 3]
```diff
@@ -71,7 +72,11 @@ class WelcomeFragment extends SSOFragment {
   }
 
   private def startCreateAccountFlow(): Unit =
-    activity.showFragment(CreateAccountFragment(), CreateAccountFragment.Tag)
+    if (BuildConfig.KOTLIN_CREATE_ACCOUNT_MIGRATION) {
```
Comment (BradleyWire) [inline app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala]: Do we need multiple build configs? The KOTLIN_MIGRATION is one project overall and when we release to production we removed the flag. I think one buildConfig flag will suffice.

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -574,4 +574,15 @@
     <dimen name="item_view_country_code_text_size">18sp</dimen>
     <dimen name="item_view_country_code_padding">24dp</dimen>
 
+
+    <!-- Create Account screen -->
+    <dimen name="create_account_layout_container_width">260dp</dimen>
```
Comment (BradleyWire) [inline app/src/main/res/values/dimens.xml]: Nice!

File: app/src/main/res-v2/layout/fragment_create_account.xml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            android:layout_gravity="center"
+            android:gravity="center"
+            android:text="@string/app_entry_create_account_sub"
+            android:textColor="@color/teams_headline_text"
+            android:textSize="@dimen/create_account_description_text_size"
+            />
+
+    </LinearLayout>
+
+    <LinearLayout
```
[the remark this answers is not in the document]
Comment (mejdoo) [reply app/src/main/res-v2/layout/fragment_create_account.xml]: ok let's do it tomorrow

File: app/src/main/res-migration/values/styles_button.xml (hunk 2 above)
Comment (mejdoo) [reply app/src/main/res-migration/values/styles_button.xml]: Possible let's discuss this with @gizemb

File: app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala (hunk 3 above)
Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala]: I don't think so. For every new Kotlin feature, we need to have a dedicated feature flag. We can decide to release one feature and later we can revert back to the scala in case of problem. @android10 what do you think ?

File: app/src/main/res/values/dimens.xml (hunk 4 above)
Comment (mejdoo) [reply app/src/main/res/values/dimens.xml]: 👍

File: app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala (hunk 3 above)
Comment (BradleyWire) [reply app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala]: But, we're not using the `KOTLIN_MIGRATION` flag in a way tradition feature flags work. You're right in that feature flags are used to release code at different times in a secure way. But, as far as this project knows we're using the `KOTLIN_MIGRATION` to switch to development of Kotlin away from Scala. 

If we're using the build system for feature flags then we'll need to create a mechanism to accommodate multiple flags for various situations when we introduce new ones. 

As a scenario that may break this or future this code is - if `KOTLIN_CREATE_ACCOUNT_MIGRATION` flag is on and `KOTLIN_MIGRATION` is OFF it'll break this feature due to the storage/dependency changes being behind `KOTLIN_MIGRATION`.  

I guess we'd need to determine if we're using the `KOTLIN_MIGRATION` for development purposes only surrounding the migration or for feature development specifically.

Comment (mejdoo) [reply app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala]: btw the current name of the buildConfig is  `KOTLIN_SETTINGS_MIGRATION` not `KOTLIN_MIGRATION` . Let's discuss this tomorrow with the team.

Comment (BradleyWire) [reply app/src/main/scala/com/waz/zclient/appentry/WelcomeFragment.scala]: We'll need to make a new flag for the core functionality in that case.

File: app/src/main/res-migration/layout/fragment_create_account.xml [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        android:focusable="true"
+        android:orientation="vertical"
+        app:layout_constraintBottom_toBottomOf="@+id/createProAccountDescriptionTextView"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/centerGuideline"
+        app:layout_constraintVertical_bias="0.1"
+        app:layout_constraintWidth_percent="0.71" />
+
+    <com.waz.zclient.ui.text.GlyphTextView
```
Comment (gizemb) [inline app/src/main/res-migration/layout/fragment_create_account.xml]: we should abandon using Scala custom views on the Kotlin side. You cannot reference them at the .kt classes, so it's a bit of a hack to include it in the corresponding xml.

Also we can directly use TextViews with drawableTop for "Personal"/"Pro" views, so it'll be a more simpler layout without glyphs. I think it's ok to use some dummy images until we get the images from the designer.

File: app/src/main/res-migration/values/styles_textview.xml [hunk 7]
```diff
@@ -3,6 +3,27 @@
 
     <style name="TextView" parent="@android:style/TextAppearance.Widget.TextView"/>
 
+    <style name="TextView.RobotoThin">
```
Comment (gizemb) [inline app/src/main/res-migration/values/styles_textview.xml]: I think it would be better to keep those styles aligned with the styleguide. What do you think about that? https://app.zeplin.io/styleguide/5de4f201d49c777852802121/textstyles

of course, it'd be better if we do it in a separate PR

File: app/src/main/res-migration/layout/fragment_create_account.xml (hunk 6 above)
Comment (mejdoo) [reply app/src/main/res-migration/layout/fragment_create_account.xml]: Ok done. I used wire icon launcher for now as a dummy image

File: app/src/main/res-migration/values/styles_textview.xml (hunk 7 above)
Comment (mejdoo) [reply app/src/main/res-migration/values/styles_textview.xml]: Why the styleguide is using `SFPro` not Roboto?  🤔

Comment (gizemb) [reply app/src/main/res-migration/values/styles_textview.xml]: that's not something that I can answer :) Maybe you can contact the designer to update them to Roboto.

Comment (mejdoo) [reply app/src/main/res-migration/values/styles_textview.xml]: We are not using this font. The styleguide should be updated I guess.

Comment (mejdoo) [reply app/src/main/res-migration/values/styles_textview.xml]: here is the current font definition in `strings.no_translate`

    ```
<!-- Typeface Definitions -->
    <string translatable="false" name="wire__glyphs" >wire_glyphs.otf</string>
    <string translatable="false" name="wire__typeface__redacted">redacted_script_regular.ttf</string>
    <string translatable="false" name="wire__typeface__light">Roboto_Light</string>
    <string translatable="false" name="wire__typeface__thin">Roboto_Thin</string>
    <string translatable="false" name="wire__typeface__regular">Roboto_Regular</string>
    <string translatable="false" name="wire__typeface__medium">Roboto_Medium</string>
    <string translatable="false" name="wire__typeface__bold">Roboto_Bold</string>
```

Comment (gizemb) [reply app/src/main/res-migration/values/styles_textview.xml]: yes, definitely. we would want the fonts to be consistent with how it has always been. we can proceed by assuming Roboto instead of SFPro maybe?

Comment (mejdoo) [reply app/src/main/res-migration/values/styles_textview.xml]: As discussed, we will leave it as it is now. The styleguide should be updated by a designer.

**Comment by mejdoo** [comment]
build again
